pkg/asset/store/store.go

```diff
@@ -125,9 +142,47 @@ func (s *storeImpl) DestroyState() error {
 	return nil
 }
 
+// validateAssets checks for expired certificates and removes the expired certificates from
+// the installer state file.
+func validateAssets(assets map[string]json.RawMessage) error {
+	for assetName, data := range assets {
+		storedAssetInfo := storedAsset{}
+		err := json.Unmarshal(data, &storedAssetInfo)
+		if err != nil {
+			logrus.Warnf("failed to unmarshal asset %s from state file: %s", assetName, err.Error())
+			continue
+		}
+
+		for _, file := range storedAssetInfo.FileList {
+			if filepath.Ext(file.Filename) == ".crt" && file.Data != nil {
```

Comment:
BTW a more generic way to implement this might be:

```
if expiringAsset, ok := storedAssetInfo.(AssetExpirer); ok {
    if err := expiringAsset.CheckExpired(); err != nil {
        return err
    }
}
```

Then the code to check for certificates expiry could live in `CertKey` instead of in the asset store.